xds/src/main/java/com/linecorp/centraldogma/xds/group/v1/XdsGroupService.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.centraldogma.xds.group.v1;
+
+import static com.linecorp.centraldogma.server.internal.admin.auth.AuthUtil.currentAuthor;
+import static com.linecorp.centraldogma.server.internal.api.RepositoryServiceUtil.createRepository;
+import static com.linecorp.centraldogma.server.internal.api.RepositoryServiceUtil.removeRepository;
+import static com.linecorp.centraldogma.xds.internal.ControlPlanePlugin.XDS_CENTRAL_DOGMA_PROJECT;
+
+import com.google.protobuf.Empty;
+
+import com.linecorp.centraldogma.common.RepositoryExistsException;
+import com.linecorp.centraldogma.server.command.CommandExecutor;
+import com.linecorp.centraldogma.server.metadata.MetadataService;
+import com.linecorp.centraldogma.server.storage.project.Project;
+import com.linecorp.centraldogma.server.storage.project.ProjectManager;
+import com.linecorp.centraldogma.xds.group.v1.XdsGroupServiceGrpc.XdsGroupServiceImplBase;
+
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.StreamObserver;
+
+/**
+ * An {@link XdsGroupServiceImplBase} implementation that provides methods to manage XDS groups.
+ */
+public final class XdsGroupService extends XdsGroupServiceImplBase {
+
+    private final ProjectManager projectManager;
+    private final CommandExecutor commandExecutor;
+    private final MetadataService mds;
+
+    /**
+     * Creates a new instance.
+     */
+    public XdsGroupService(ProjectManager projectManager, CommandExecutor commandExecutor) {
+        this.projectManager = projectManager;
+        this.commandExecutor = commandExecutor;
+        mds = new MetadataService(projectManager, commandExecutor);
+    }
+
+    @Override
+    public void createGroup(CreateGroupRequest request,
+                            StreamObserver<Group> responseObserver) {
+        final String groupName = request.getGroup().getName();
+        final String name = removePrefix("groups/", groupName);
```

Comment:
Taking another look at this PR, it seems like we're going need to extra logic to remove/add `groups/` in all user-facing locations. i.e. API paths, DTOs, UI, metrics, etc.. I'm not sure if it's realistic and worth it to ensure this.

Do you mind explaining why we need this prefix again?

Reply:
Like I said before, I'm trying to adhere to the AIP as much as possible:
https://google.aip.dev/122
It recommends in this way:
```
publishers/123/books/les-miserables
users/vhugo1802

Resource name components should usually alternate between collection identifiers (example: publishers, books, users) and resource IDs (example: 123, les-miserables, vhugo1802).
```

>  we're going need to extra logic to remove/add groups/ in all user-facing locations.

Yeah, that is true. But the logic isn't so many, and we still need that logic for validating and removing group name from the cluster name.
I thought the format (`groups/{group}/clusters/{cluster}`), which follows the guideline, was much better than the original format (`{group}/{cluster}`).

Reply:
If you're strong on following the AIP, then sure. Ideally I prefer that there is a cross-cutting solution to apply this, but I guess we'll be doing this manually then.

e.g. If a new API is added where groupName is the parameter, we should be conscious to add/remove the prefix.

Reply:
>  Ideally I prefer that there is a cross-cutting solution to apply this

I also prefer applying the cross-cutting solution, which is why I hesitated to follow the AIP at first. 😕 However, I've decided to try using gRPC for this API, and it’s not easy to apply AOP even if I don’t follow AIP because it needs parameters in the message. So, I've decided to just follow the AIP.